packages/vitest/src/runtime/worker.ts

```diff
@@ -1,94 +1,55 @@
-import { performance } from 'node:perf_hooks'
-import { createBirpc } from 'birpc'
 import { workerId as poolId } from 'tinypool'
-import type { CancelReason } from '@vitest/runner'
-import type { RunnerRPC, RuntimeRPC, WorkerContext, WorkerGlobalState } from '../types'
-import { getWorkerState } from '../utils/global'
+import { ModuleCacheMap } from 'vite-node/client'
+import type { ContextRPC } from '../types/rpc'
 import { loadEnvironment } from '../integrations/env/loader'
-import { mockMap, moduleCache, startViteNode } from './execute'
+import type { WorkerGlobalState } from '../types/worker'
 import { setupInspect } from './inspector'
-import { createSafeRpc, rpcDone } from './rpc'
+import { createRuntimeRpc, rpcDone } from './rpc'
+import type { VitestWorkerConstructor } from './workers/types'
 
-async function init(ctx: WorkerContext) {
-  // @ts-expect-error untyped global
-  const isInitialized = typeof __vitest_worker__ !== 'undefined'
-  const isIsolatedThreads = ctx.config.pool === 'threads' && (ctx.config.poolOptions?.threads?.isolate ?? true)
+export async function run(ctx: ContextRPC) {
```

Comment:
Would this file become part of public API for custom pool implementers? 
They would be providing their pool which loads this file and spins up the runner?

Reply:
Yes. Ideally, I just wanted to have a single tinypool instance and other pools would just provide a different `.worker` value. But right now I guess we can just expose this